static/js/components/TableColumnsSelect.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import {
+  CheckboxInput,
+  ContextualMenu,
+  Icon,
+} from "@canonical/react-components";
+import classnames from "classnames";
+
+interface Props {
+  className?: string;
+  columns: string[];
+  hidden: string[];
+  setHidden: (columns: string[]) => void;
+}
+
+const TableColumnsSelect = ({
+  className,
+  columns,
+  hidden,
+  setHidden,
+}: Props): JSX.Element => {
+  const selectedCount = columns.length - hidden.length;
+
+  const toggleHiddenColumn = (column: string): void => {
+    if (hidden.includes(column)) {
+      setHidden(hidden.filter((c) => c !== column));
+    } else {
+      setHidden([...hidden, column]);
+    }
+  };
+  return (
+    <ContextualMenu
+      className={classnames(className, "configurable-table-toggle")}
+      dropdownProps={{ "aria-label": "columns menu" }}
+      position="right"
+      toggleClassName="has-icon"
+      toggleLabel={<Icon name="settings" />}
+      toggleAppearance="base"
+    >
+      <div className="table-column-select-list">
+        <CheckboxInput
+          checked={hidden.length === 0}
+          indeterminate={selectedCount > 0 && selectedCount < columns.length}
+          label={`${selectedCount} out of ${columns.length} selected`}
+          onChange={() => hidden.length > 0 && setHidden([])}
```

Comment:
I find this a bit confusing. What I would expect here is: if I click this checkbox when some of the checkboxes below are deselected, they turn to "all selected" (and this works), and if I click this checkbox when all checkboxes below are selected, they get all deselected - but this doesn't happen.

Reply:
Adjusted it so it matches expected behaviour.